### DISCUSSION TITLE
Huge amount of error in VS

### DISCUSSION THREAD

**Discussion by MarkingN1** [YimMenu/YimMenu#1715] [Q&A]
I have this problem where i did everything as the tutorial said and with Yimmenu i have no kidding 1440 error in VS. I downloaded Tupoyemenu and i had no issue at all. I would really like to build the menu myself, work on it, try out new things and just learn about codeing. I hope someone could help.
![sok error](https://github.com/YimMenu/YimMenu/assets/137184233/53d171fa-7adc-42b9-b98a-7e11639a4e91)

**Comment by xiaoxiao921** [comment]
Did you try using the Visual Studio CMake extension?

https://github.com/YimMenu/YimMenu/wiki/Setup-your-PC-for-YimMenu-Development#compiling

**Reply by MarkingN1** [reply]
Yup i did everything as the tutorial and it worked with Tupoye but for some reason with Yimmenu i have 1440 error.

**Comment by MarkingN1** [comment]
Ok guys my Mingw compiler was bad that caused the problem (#error: no supported target architecture C:/..../winnt.h). It took me like 5 days of installing and reinstalling everything and doing research about VS and compilers.TY

**Reply by Yimura** [reply]
YimMenu doesn't support compilation under MingW, mainly because of compiler specific implementations on how vtables are implemented.

Because GTA 5 is compiled with MSVC we need to match their way of structuring vtable in memory. If we use a different compiler this will cause issues and requires manual intervention to fix.

**Reply by tupoy-ya** [reply]
I'd say the main reason we don't support MinGW is that #1227 uses a C++ 23 feature that is not implemented outside of MSVC.

The vtable problems can be ifdefed with enough testing.

**Reply by Yimura** [reply]
Also true, kinda forgot about that. However with further development of the base we'll probably stray further and further away from supporting MingW.

I've recently experimented on Linux and my current setup is as follows, setup a Windows VM, follow the install instructions from the Wiki.
Then setup an SSH server on the VM so I can remote into it. From there you can use Visual Studio Code and use the remote explorer to connect over SSH to your VM.

Visual Studio Code will then threat this as if you were directly developing on the machine, no need for RDP or shitty VM screens to code on there. Just install the C++ extensions from the wiki on your remote and you should be good to go.

**Reply by xiaoxiao921** [reply]
> I'd say the main reason we don't support MinGW is that #1227 uses a C++ 23 feature that is not implemented outside of MSVC.
> 
> The vtable problems can be ifdefed with enough testing.

I already gave you a godbolt link which compiles on all majir compilers, the only msvc specific is the usage of const char* which can be replaced by a compile time str struct which is defined like in common.hpp

also for vtable it should be good now that RTTI is not used anymore by gta5
